### PR TITLE
UIOR-444 initial commit

### PR DESCRIPTION
**Pull request by aliaksei-chumakou**
## Purpose
Ability to change quantity ordered without closing order
https://issues.folio.org/browse/UIOR-444

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/LayerCollection/LayerPOLine.js [hunk 1]
```diff
@@ -104,6 +107,8 @@ function LayerPOLine({
       if (response.errors && response.errors.length) {
         if (response.errors.find(el => el.code === 'lines_limit')) {
           openLineLimitExceededModal(line);
+        } else if (response.errors.find(el => el.code === 'piecesNeedToBeDeleted')) {
```
Comment (AndreiShumski2) [inline src/components/LayerCollection/LayerPOLine.js:110]: `ERROR_CODES.piecesNeedToBeDeleted`

File: src/components/ModalDeletePieces/ModalDeletePiecesContainer.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        batchFetch(mmutator.linePieces, poLines, (itemsChunk) => {
+          const query = itemsChunk
+            .map(({ id }) => `poLineId==${id}`)
+            .join(' or ');
+
+          return query ? `(${query}) sortby locationId` : '';
+        }, {})
+          .then(piecesResp => getHydratedPieces(piecesResp, mmutator.requests, mmutator.items, mmutator.pieceLocations))
+          .then(setPieces)
+          .then(setIsLoading);
```
Comment (NikitaSedyx) [inline src/components/ModalDeletePieces/ModalDeletePiecesContainer.js:38]: does it set result of `setPieces`? what does it return?

Comment (aliaksei-chumakou) [reply src/components/ModalDeletePieces/ModalDeletePiecesContainer.js:38]: could you please rephrase the question? setIsLoading doesn't set any result, it turns off the loading spinner

Comment (aliaksei-chumakou) [reply src/components/ModalDeletePieces/ModalDeletePiecesContainer.js]: changed to more explicit way

**Pull request merged**